fawltydeps/packages.py

```diff
@@ -317,14 +317,27 @@ class TemporaryPipInstallResolver(BasePackageResolver):
     def installed_requirements(
         venv_dir: Path, requirements: List[str]
     ) -> Iterator[Path]:
-        """Install the given requirements into venv_dir with `pip install`."""
+        """Install the given requirements into venv_dir with `pip install`.
+
+        If `pip install` fails, we will log a warning, then retry by attempting
+        to `pip install` each given requirement in order (logging any errors
+        that occur), before we finally enter the caller's context.
+
+        It is up to the caller to detect/handle any requirements that we failed
+        to install.
+        """
```

Comment:
```suggestion
        It is up to the caller to handle any requirements that we failed
        to install.
        """
```
Nit: detection is now done via looping over all `requirements`.